Adding Auth0 Security Bulletin: Assigning Scopes Based on Email Address (#8205)
auth0/docs: articles/security/bulletins/2019-09-05_scopes.md

---
title: Auth0 Security Bulletin for Assigning Scopes Based on Email Address
description: Cases where improper custom rule code may create vulnerabilities in the authentication flow.
topics:
  - security
  - security-bulletins
  - rules
contentType:
  - reference
useCase:
  - development
---

# Auth0 Security Bulletin for Assigning Scopes Based on Email Address

## Overview

If you use rules to assign scopes to users based on their email address there is a possibility that scopes could be compromised if your application uses multiple connections.

Auth0 requires that email addresses are unique per connection, but not per application (client). Therefore, if you are using rules to assign scopes based on email address, and if a legitimate user A (`user_a@domain.com`) signs up via one connection, and a malicious user B signs up on a different connection using legitimate user A's email address, then malicious user B will get the same scopes as legitimate user A. This would result in two distinct users with the same email address and same scopes.

## Am I affected?

> If you are using rules to assign scopes to users based on email address and you are using multiple connections per application, you may be vulnerable.

Reviewer: ```suggestion
You are vulnerable if:

* You are using rules to assign scopes to users based on email address **and**
* You are using multiple connections per application
```

## How do I fix this?

> In this scenario, the most straightforward mitigation is to require that users [verify their email address](/api/management/v2#!/Jobs/post_verification_email) before being able to complete the login process.

Reviewer: ```suggestion
The most straightforward mitigation is to require users to [verify their email address](/api/management/v2#!/Jobs/post_verification_email) after signing up and before being allowed to log in.
```

---
title: Auth0 Security Bulletin for Assigning Scopes Based on Email Address
description: Cases where improper custom rule code may create vulnerabilities in the authentication flow.
topics:
  - security
  - security-bulletins
  - rules
contentType:
  - reference
useCase:
  - development
---

# Auth0 Security Bulletin for Assigning Scopes Based on Email Address

## Overview

If you:

> * Use rules to assign scopes to users based on their email addresses

Reviewer: ```suggestion
* Use rules to assign scopes to users based on their email addresses **and**
```

Reviewer: Only change here, @jeffreylees so that people understand it's not either/or -- it's both; what do you think?
